stock_split_picking/wizards/stock_split_picking.py

```diff
@@ -43,6 +52,23 @@ def _apply_move(self):
                 new_pickings += picking._split_off_moves(move)
         return self._picking_action(new_pickings)
 
+    def _apply_available_line(self):
+        """Create different pickings for available and not available move line"""
+        for picking in self.mapped("picking_ids"):
+            if picking.picking_type_code == "outgoing":
+                moves = picking.mapped("move_ids")
+                moves_available = moves.filtered(lambda move: move.state == "confirmed")
+                new_picking = moves_available.mapped("picking_id")._split_off_moves(
+                    moves_available
+                )
+            else:
+                raise UserError(_("This type of mode is not available for the receipt"))
+        return self._picking_action(new_picking)
+
+    def _apply_available_product(self):
+        """Create different pickings for available and not available move line"""
+        return self.mapped("picking_ids").split_process("reserved_availability")
```

Comment:
After splitting a picking based on available product, the picking that contain all the product available is still in the state confirm (user need to click on check availability), because all the line that was partially available are still partially available (the state is not recomputed automatically).
So after calling the split_process you need to call "_recompute_state" on the move
```suggestion
        self.picking_ids.split_process("reserved_availability")
        self.picking_ids.move_ids._recompute_state()
        return True
        
```